src/Particle/Par_Init_ByFile.cpp

```diff
@@ -129,13 +133,30 @@ void Par_Init_ByFile()
 //    collect data for the target particle
 //    [id][att]
       if ( amr->Par->ParICFormat == PAR_IC_FORMAT_ID_ATT )
-         memcpy( ParData1, ParData_ThisRank+p*NParAtt, NParAtt*sizeof(real) );
+      {
+         if ( sizeof(real) == load_data_size )
+            memcpy( ParData1, ParData_ThisRank+p*NParAtt*load_data_size, NParAtt*load_data_size );
+         else
+         {
+            if ( PAR_IC_FLOAT8 )
+               for (int v=0; v<NParAtt; v++) ParData1[v] = (real)( *((double*)(ParData_ThisRank+(p*NParAtt+v)*load_data_size)) );
+            else
+               for (int v=0; v<NParAtt; v++) ParData1[v] = (real)( *((float* )(ParData_ThisRank+(p*NParAtt+v)*load_data_size)) );
+         }
+      }
 
 //    [att][id]
       else
       {
-         for (int v=0; v<NParAtt; v++)
-            ParData1[v] = ParData_ThisRank[ v*NParThisRank + p ];
+         if ( sizeof(real) == load_data_size )
+            for (int v=0; v<NParAtt; v++) ParData1[v] = *((real*)(ParData_ThisRank+(v*NParThisRank+p)*load_data_size));
+         else
```

Comment:
It seems redundant. Can we remove it and rely on `if ( PAR_IC_FLOAT8 ) ... else ...`?

Reply:
Yes, I think we can 😅 .